packages/text-bitmap/src/BitmapFont.ts

```diff
@@ -226,4 +236,298 @@ export class BitmapFont
+     * @param {string|string[]} options.chars - characters included in the font set. You can also
+     *      use ranges. For example, `[['a', 'z'], ['A', 'Z'], "!@#$%^&*()~`{}[]"]. Don't forget to
+     *      include spaces ' ' in your character set!
+     * @param {number} [options.resolution=1] - optional resolution to render the glyphs at
+     * @return {PIXI.BitmapFont} font generated by style options.
+     * @static
+     * @example
+     * const font = PIXI.BitmapFont.from({
+     *     chars: [['a', 'z'], ['A', 'Z'], "!#$%^&*()~`{}[]"]
+     *     name: "TitleFont", // optional
+     *     fontFamily: "Arial",
+     *     fontSize: 12,
+     *     strokeThickness: 2,
+     *     fill: "purple"
+     * });
+     *
+     * const text = new PIXI.BitmapText("This is an example using Pixi's BitmapText+Font", font);
+     *
+     * const title = new PIXI.BitmapText("This is the title", { font: "TitleFont" });
+     */
+    public static from(options: IBitmapFontFactoryOptions): BitmapFont
+    {
+        const name = options.name;
+        let chars = options.chars;
+
+        if (name && BitmapFont.available[name])
+        {
+            throw new Error(`[Name Collision]: The name "${name}" has already been taken by another bitmap font`);
+        }
+
+        // Split the chars string into individual characters
+        if (typeof chars === 'string')
+        {
+            chars = chars.split('');
+        }
+        // Handle an array of characters+ranges
+        else if (chars.find((elem) => Array.isArray(elem)))
+        {
+            const flatChars = [];
+
+            for (let i = 0, j = chars.length; i < j; i++)
+            {
+                const elem = chars[i];
+
+                // Handle range delimited by start/end chars
+                if (Array.isArray(elem))
+                {
+                    if (elem.length !== 2)
+                    {
+                        let suffix;
+
+                        switch (i)
+                        {
+                            case 1: suffix = 'st'; break;
+                            case 2: suffix = 'nd'; break;
+                            case 3: suffix = 'rd'; break;
+                            default: suffix = 'th';
+                        }
+
+                        throw new Error('BitmapFont.from was provided a character range is more than two limits. '
+                            + `This is the ${i}${suffix} input in the character set.`);
+                    }
+
+                    for (let i = elem[0].charCodeAt(0), j = elem[1].charCodeAt(0); i <= j; i++)
+                    {
+                        flatChars.push(String.fromCharCode(i));
+                    }
+                }
+                // Handle a character set string
+                else
+                {
+                    flatChars.push(...elem.split(''));
+                }
+            }
+
+            chars = flatChars;
```

Comment:
My other concern, that'd be nice to be at least known with some sort of debug flag, or logging, is when BitmapText is going to try to draw a character that is _not_ listed. I'm going to be using this for all sorts of fast counting up currency meters and such, so what if I left out a Euro symbol, by accident? Currently, it'll just silently miss out that char when rendering, but I'd really like to know that I missed something

Reply:
This is BitmapText behavior and probably another PR. Since this PR doesn’t touch BitmapText and only uses it for rendering.